pkg/executables/cmk.go

```diff
@@ -195,39 +187,34 @@ func (c *Cmk) ValidateAffinityGroupsPresent(ctx context.Context, domainId string
 	return nil
 }
 
-func (c *Cmk) ValidateZonesPresent(ctx context.Context, zones []v1alpha1.CloudStackZone) ([]v1alpha1.CloudStackResourceIdentifier, error) {
-	var zoneIdentifiers []v1alpha1.CloudStackResourceIdentifier
-	for _, zone := range zones {
-		command := newCmkCommand("list zones")
-		if len(zone.Id) > 0 {
-			applyCmkArgs(&command, withCloudStackId(zone.Id))
-		} else {
-			applyCmkArgs(&command, withCloudStackName(zone.Name))
-		}
-		result, err := c.exec(ctx, command...)
-		if err != nil {
-			return nil, fmt.Errorf("getting zones info - %s: %v", result.String(), err)
-		}
-		if result.Len() == 0 {
-			return nil, fmt.Errorf("zone %s not found", zone)
-		}
+func (c *Cmk) ValidateZonePresent(ctx context.Context, zone v1alpha1.CloudStackZone) (string, error) {
+	command := newCmkCommand("list zones")
+	if len(zone.Id) > 0 {
+		applyCmkArgs(&command, withCloudStackId(zone.Id))
+	} else {
+		applyCmkArgs(&command, withCloudStackName(zone.Name))
+	}
+	result, err := c.exec(ctx, command...)
+	if err != nil {
+		return "", fmt.Errorf("getting zones info - %s: %v", result.String(), err)
+	}
+	if result.Len() == 0 {
+		return "", fmt.Errorf("zone %s not found", zone)
+	}
 
-		response := struct {
-			CmkZones []cmkResourceIdentifier `json:"zone"`
-		}{}
-		if err = json.Unmarshal(result.Bytes(), &response); err != nil {
-			return nil, fmt.Errorf("parsing response into json: %v", err)
-		}
-		cmkZones := response.CmkZones
-		if len(cmkZones) > 1 {
-			return nil, fmt.Errorf("duplicate zone %s found", zone)
-		} else if len(zones) == 0 {
-			return nil, fmt.Errorf("zone %s not found", zone)
-		} else {
-			zoneIdentifiers = append(zoneIdentifiers, v1alpha1.CloudStackResourceIdentifier{Name: cmkZones[0].Name, Id: cmkZones[0].Id})
-		}
+	response := struct {
+		CmkZones []cmkResourceIdentifier `json:"zone"`
+	}{}
+	if err = json.Unmarshal(result.Bytes(), &response); err != nil {
+		return "", fmt.Errorf("parsing response into json: %v", err)
+	}
+	cmkZones := response.CmkZones
+	if len(cmkZones) > 1 {
+		return "", fmt.Errorf("duplicate zone %s found", zone)
+	} else if len(cmkZones) == 0 {
+		return "", fmt.Errorf("zone %s not found", zone)
 	}
-	return zoneIdentifiers, nil
+	return cmkZones[0].Id, nil
```

Comment:
The input argument in this function is the name of a zone. This function resolves the name into the ID while validating the zone's presence by name.

Reply:
It sounds like this would be better named `ValidateAndGetZoneId` especially because the other Validate methods only return an error, but yea I get it now.